### PR TITLE
026 - recommendation fetch API / implementation

### PR DESCRIPTION
**Pull request by garySiladi**
This PR implements the recommended fetch and also makes it visible on the homepage.

### CODE REVIEW

File: src/api/fetch.js [hunk 1]
```diff
@@ -9,6 +9,15 @@ function constructFetch(type: string, id: ?number) {
   .catch(err => err);
 }
 
+export function getRecommendedEpisodes(id: number) {
```
Comment (varholak-peter) [inline src/api/fetch.js]: rename to `userId`

File: src/components/app/app.js [hunk 2]
```diff
@@ -117,6 +117,20 @@ class App extends React.Component { // eslint-disable-line react/prefer-stateles
       selectedEpisode: 0,
     });
   }
+  combineSeries(dataSeries: Object, dataRecommended: Object) {
```
Comment (varholak-peter) [inline src/components/app/app.js]: I would recommend making this a pure function which returns combined series (also change name in that case).

File: src/components/app/app.js [hunk 3]
```diff
@@ -29,17 +30,16 @@ class App extends React.Component { // eslint-disable-line react/prefer-stateles
...
     goToEpisode: boolean,
   }
-  componentWillMount() {
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyPress);
     getRoot()
-    .then((data) => {
-      this.setState({
-        series: data.shelves,
+    .then((dataSeries) => {
```
Comment (varholak-peter) [inline src/components/app/app.js]: This waits until one finishes to make a second request, consider making two separate requests and check if the data of the other one was set to combine them.

File: src/components/app/app.js [hunk 4]
```diff
@@ -117,6 +117,20 @@ class App extends React.Component { // eslint-disable-line react/prefer-stateles
...
     });
   }
+  combineSeries(dataSeries: Object, dataRecommended: Object) {
+    const shelves = dataSeries.shelves || [];
+    const recommendedSlider = {
+      title: 'Recommended',
+      items: dataRecommended.recommended_videos,
+      type: 'series',
+    };
+    shelves.splice(0, 0, recommendedSlider);
```
Comment (varholak-peter) [inline src/components/app/app.js]: you might want to use concat in order to not modify the current array

File: src/components/app/app.js (hunk 3 above)
Comment (garySiladi) [inline src/components/app/app.js]: yep, you're right

File: src/components/app/app.js (hunk 4 above)
Comment (garySiladi) [inline src/components/app/app.js]: I will look into it

File: src/api/fetch.js (hunk 1 above)
Comment (garySiladi) [inline src/api/fetch.js]: ok

File: src/components/app/app.js [hunk 5]
```diff
@@ -29,16 +41,22 @@ class App extends React.Component { // eslint-disable-line react/prefer-stateles
...
+        ),
+      });
+    });
+    getRecommendedEpisodes(1)
+    .then((dataRecommended) => {
+      this.setState({
+        series: [App.createRecommendedSlider(dataRecommended)].concat(this.state.series),
       });
     });
-  }
```
Comment (varholak-peter) [inline src/components/app/app.js]:
```
combineSeries() {
  this.setState({ series: this.state.recommendedSeries.concat(this.state.rootSeries), });
}

componentDidMount() {
  document.addEventListener('keydown', this.handleKeyPress);
  getRoot()
  .then((data) => {
    this.setState({
      rootSeries: data.shelves,
    });
    if (recommendedSeries) {
      combineSeries();
    }
  });
  getRecommendedEpisodes(1)
  .then((dataRecommended) => {
    this.setState({
      recommendedSeries: dataRecommended.series,
    });
    if (rootSeries) {
      combineSeries();
    }
  });
}
```

**Pull request merged**